llvm/lib/Target/X86/X86PfmCounters.td

```diff
@@ -121,7 +121,12 @@ def HaswellPfmCounters : ProcPfmCounters {
     PfmIssueCounter<"HWPort7", "uops_executed_port:port_7">
   ];
   let ValidationCounters = [
-    PfmValidationCounter<InstructionRetired, "INSTRUCTIONS_RETIRED">
+    PfmValidationCounter<InstructionRetired, "INSTRUCTIONS_RETIRED">,
+    PfmValidationCounter<L1DCacheLoadMiss, "MEM_LOAD_UOPS_RETIRED:L1_MISS">,
+    PfmValidationCounter<L1ICacheLoadMiss, "L1-ICACHE-LOAD-MISSES">,
+    PfmValidationCounter<DataTLBLoadMiss, "DTLB_LOAD_MISSES:MISS_CAUSES_A_WALK">,
+    PfmValidationCounter<DataTLBStoreMiss, "DTLB_STORE_MISSES:MISS_CAUSES_A_WALK">,
+    PfmValidationCounter<InstructionTLBLoadMiss, "ITLB_MISSES:MISS_CAUSES_A_WALK">
   ];
```

Comment:
Could we pull this out into a default list instead of duplicating it? `let ValidationCounters = DefaultX86ValidationCounters` or something?

Reply:
Thanks for the suggestion! I've refactored everything into a default counters set that's split across AMD/Intel (as the counters between them differ). We might need to adapt as newer/older architectures are brought up, but this should be reasonably scalable.